### PR TITLE
enable logical assignment in babel preset env

### LINKED ISSUE #11861: Enable logical assignment in @babel/preset-env
## Feature Request

<div align="center">

### Logical assignment is Stage 4!

</div>

This means that it should be enabled by default in `@babel/preset-env` :tada:

It should be done similarly to this commit: https://github.com/babel/babel/commit/b92ad318f173e1a0573a188324aa0c222ea35782

1. Add it to `packages/babel-preset-env/src/available-plugins.js`
2. Add the `compat-table` mapping to `packages/babel-compat-data/scripts/data/plugin-features.js` (you can find the correct feature name at https://kangax.github.io/compat-table/esnext/)
3. Run `cd packages/babel-compat-data` and `yarn build-data` to update the data about the supported browser versions
4. Update the test as needed (you don't need to write new tests)
5. PR! :tada:

---

If it is the first time that you contribute to Babel, follow these steps: (you need to have `make` and `yarn` available on your machine)
1. **Write a comment there** to let other possible contributors know that you are working on this bug.
1. Fork the repo
1. Run `git clone https://github.com/<YOUR_USERNAME>/babel.git && cd babel`
1. Run `yarn && make bootstrap`
1. Wait :hourglass_flowing_sand: 
1. Run `make watch` (or `make build` whenever you change a file)
1. Add a test if needed (only `input.js`; `output.js` will be automatically generated)
1. Update the code!
1. `yarn jest babel-preset-env` to run the tests
    - If some test outputs don't match but the new results are correct, you can delete the bad `output.js` files and run the tests again
    - If you prefer, you can run `OVERWRITE=true yarn jest babel-preset-env` and they will be automatically updated.
1. If it is working, run `make test` to run all the tests
1. Run `git push` and open a PR!

### PR DESCRIPTION
**Pull request by morrme**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #11861 
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR Link    | n/a 
| Any Dependency Changes?  | No
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/11876"></a>

### CODE REVIEW

File: packages/babel-preset-env/src/available-plugins.js [hunk 1]
```diff
@@ -17,6 +18,7 @@ import proposalClassProperties from "@babel/plugin-proposal-class-properties";
 import proposalDynamicImport from "@babel/plugin-proposal-dynamic-import";
 import proposalExportNamespaceFrom from "@babel/plugin-proposal-export-namespace-from";
 import proposalJsonStrings from "@babel/plugin-proposal-json-strings";
+import proposalLogicalAssignmentOperators from "@babel/plugin-proposal-logical-assignment-operators";
```
Comment (nicolo-ribaudo) [inline packages/babel-preset-env/src/available-plugins.js:21]: Can you also add these two plugins to the dependencies in package.json?

**Comment by morrme** [comment]
> Can you update the test fixtures?
> You can run
> 
> ```
> OVERWRITE=true yarn jest
> ```

`yarn` seems to be giving me errors, I will see if I can sort them out to make this work.

File: packages/babel-compat-data/data/plugins.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "samsung": "11",
     "electron": "6"
   },
+  "proposal-logical-assignment-operators": {},
```
Comment (nicolo-ribaudo) [inline packages/babel-compat-data/data/plugins.json:24]: Note (not in this PR): we probably need to update compat-table

**Comment by existentialism** [comment]
@morrme thanks!

**Pull request merged**